scaffoldmaker/meshtypes/meshtype_3d_haustra1.py

```diff
@@ -0,0 +1,426 @@
+            'Use cross derivatives',
+            'Use linear through wall',
+            'Refine',
+            'Refine number of elements around',
+            'Refine number of elements along haustrum',
+            'Refine number of elements through wall'
+        ]
+
+    @staticmethod
+    def checkOptions(options):
+        for key in [
+            'Number of elements through wall',
+            'Refine number of elements around',
+            'Refine number of elements along haustrum',
+            'Refine number of elements through wall']:
+            if options[key] < 1:
+                options[key] = 1
+        if (options['Number of elements around'] < 9) :
+            options['Number of elements around'] = 9
+        if (options['Number of elements around'] % 3 != 0) :
+            options['Number of elements around'] = (options['Number of elements around']//3)*3
+        if (options['Number of elements along haustrum'] < 2) :
+            options['Number of elements along haustrum'] = 2
+        for key in [
+            'Inner radius',
+            'Haustra inner radius factor',
+            'Haustrum segment end-derivative factor',
+            'Haustrum segment mid-derivative factor',
+            'Wall thickness',
+            'Haustra segment length']:
+            if options[key] < 0.0:
+                options[key] = 0.0
+        if options['Corner inner radius factor'] < 0.1:
+            options['Corner inner radius factor'] = 0.1
+        for key in [
+            'Corner inner radius factor',
+            'Haustrum segment end-derivative factor']:
+            if options[key] > 1.0:
+                options[key] = 1.0
+
+    @staticmethod
+    def generateBaseMesh(region, options):
+        """
+        Generate the base tricubic Hermite mesh. See also generateMesh().
+        :param region: Zinc region to define model in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        :return: None
+        """
+        elementsCountAround = options['Number of elements around']
+        elementsCountAlongHaustrum = options['Number of elements along haustrum']
+        elementsCountThroughWall = options['Number of elements through wall']
+        radius = options['Inner radius']
+        cornerInnerRadiusFactor = options['Corner inner radius factor']
+        haustraInnerRadiusFactor = options['Haustra inner radius factor']
+        haustrumSegmentEndDerivativeFactor = options['Haustrum segment end-derivative factor']
+        haustrumSegmentMidDerivativeFactor = options['Haustrum segment mid-derivative factor']
+        wallThickness = options['Wall thickness']
+        haustraSegmentLength = options['Haustra segment length']
+        useCrossDerivatives = options['Use cross derivatives']
+        useCubicHermiteThroughWall = not(options['Use linear through wall'])
+        elementsCountAlong = elementsCountAlongHaustrum
+
+        nodeIdentifier = 1
```

Comment:
The remainder of this function should call getColonHaustraSegmentInnerPoints and then generate with the tube mesh - copy the code from 'Colon 1'.
You just have to make a centre line:
x = [  [  0.0, 0.0, 0.0 ], [ haustrumLength, 0.0, 0.0 ] ]
d1 = [  [ haustrumLength, 0.0, 0.0 ], [ haustrumLength, 0.0, 0.0 ]  ]